[RHACS] [Docs] ROX-29351: Adding documentation for keyless verification (#94078)
openshift/openshift-docs: modules/verifying-image-signatures-to-ensure-image-integrity.adoc

// Module included in the following assemblies:
//
// * operating/verify-image-signatures.adoc

:_mod-docs-content-type: PROCEDURE
[id="verifying-image-signatures-to-ensure-image-integrity_{context}"]
= Verifying image signatures to ensure image integrity

> By scanning an image signature, you can verify that a trusted source signs the container image and the scan output includes a valid transparency log data to ensure image integrity.

Reviewer: ```suggestion
By scanning an image signature, you can verify that a trusted source signed the container image and the scan output includes a valid transparency log bundle to ensure image integrity.
```

> By scanning an image signature, you can verify that a trusted source signed the container image and the scan output includes a valid transparency log bundle to ensure image integrity.

Reviewer: I think we should mention a tricky edge case here. When the image in question is a multi-arch image, one has to be extra careful with signing the "correct" digest. That is because multi-arch images are presented by an index image (identified by the index digest), that acts a container for the architecture specific manifests (e.g. `amd64` and `arm64`). The latter then make up the architecture specific images (each identified by the architecture digest). Now in this case, we recommend signing both the "outer" index digest and all the "inner" architecture specific digests. This avoid problems with images resolution by the container runtime.

.Prerequisites

* You have created a signature integration.
+
For more information about how to create a signature integration, see "Securing container Images by using signature integration". 

.Procedure

* To scan an image signature by using the `roxctl` CLI, run the following command:
+
[source,terminal]
----
$ roxctl image scan \

> --image=<registry>/<repository>/<image>:<tag> \

Reviewer: ```suggestion
--image=<registry>/<repository>/<image>@<digest> \
```
Prefer using digest over tag.

--force-insecure-skip-tls-verify
----
+
where:
+
--
`<registry>`:: Specifies the container image registry. For example, `quay.io`.
`<repository>`:: Specifies the namespace that owns the image. For example, `stehessel`.
`<image>`:: Specifies the name of the container image that you want to scan. For example, `busybox`.
`<tag>`:: Specifies the name of the signature integration. For example, `sigstore`.
--
+

> Verify that the output displays the signature with the Rekor bundle and the certificate verification data. You must capture the Rekor bundle which includes the proof of inclusion and transparency log.

Reviewer: ```suggestion
Verify that the output displays the signature with the Rekor bundle and the certificate verification data. You must capture the Rekor bundle which includes the proof of inclusion into the transparency log.
```
